experiments/AtmosLES/bomex.jl

```diff
@@ -0,0 +1,511 @@
+  # Accumulate sources
+  source.ρu -= fkvector × (state.ρu .- state.ρ*u_geo)
+  return nothing
+end
+
+"""
+  Bomex Sponge (Source)
+"""
+struct BomexSponge{FT} <: Source
+  "Maximum domain altitude (m)"
+  z_max::FT
+  "Altitude at with sponge starts (m)"
+  z_sponge::FT
+  "Sponge Strength 0 ⩽ α_max ⩽ 1"
+  α_max::FT
+  "Sponge exponent"
+  γ::FT
+  "Eastward geostrophic velocity `[m/s]` (Base)"
+  u_geostrophic::FT
+  "Eastward geostrophic velocity `[m/s]` (Slope)"
+  u_slope::FT
+  "Northward geostrophic velocity `[m/s]`"
+  v_geostrophic::FT
+end
+function atmos_source!(s::BomexSponge, atmos::AtmosModel, source::Vars, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+
+  z_max = s.z_max
+  z_sponge = s.z_sponge
+  α_max = s.α_max
+  γ = s.γ
+  u_geostrophic = s.u_geostrophic
+  u_slope       = s.u_slope
+  v_geostrophic = s.v_geostrophic
+
+  z          = altitude(atmos.orientation,aux)
+  u_geo      = SVector(u_geostrophic + u_slope * z, v_geostrophic, 0)
+  ẑ          = vertical_unit_vector(atmos.orientation, aux)
+  # Accumulate sources
+  if z_sponge <= z
+    r = (z - z_sponge)/(z_max-z_sponge)
+    β_sponge = α_max * sinpi(r/2)^s.γ
+    source.ρu -= β_sponge * (state.ρu .- state.ρ * u_geo)
+  end
+  return nothing
+end
+
+"""
+  BomexTendencies (Source)
+Moisture, Temperature and Subsidence tendencies
+"""
+struct BomexTendencies{FT} <: Source
+  "Advection tendency in total moisture `[s⁻¹]`"
+  ∂qt∂t_peak::FT
+  "Lower extent of piecewise profile (moisture term) `[m]`"
+  zl_moisture::FT   
+  "Upper extent of piecewise profile (moisture term) `[m]`"
+  zh_moisture::FT
+  "Cooling rate `[K/s]`"
+  ∂θ∂t_peak::FT
+  "Lower extent of piecewise profile (subsidence term) `[m]`"
+  zl_sub::FT
+  "Upper extent of piecewise profile (subsidence term) `[m]`"
+  zh_sub::FT
+  "Subsidence peak velocity"
+  w_sub::FT
+  "Max height in domain"
+  z_max::FT
+end
+function atmos_source!(s::BomexTendencies, atmos::AtmosModel, source::Vars, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+  FT    = eltype(state)
+  ρ     = state.ρ
+  z     = altitude(atmos.orientation,aux)
+  q_tot = state.moisture.ρq_tot / state.ρ
+  TS    = thermo_state(atmos.moisture, atmos.orientation, state, aux)
+  
+  # Moisture tendencey (sink term) 
+  # Temperature tendency (Radiative cooling)
+  # Large scale subsidence
+  # Unpack struct
+  zl_moisture = s.zl_moisture
+  zh_moisture = s.zh_moisture
+  z_max       = s.z_max
+  zl_sub      = s.zl_sub
+  zh_sub      = s.zh_sub
+  w_sub       = s.w_sub
+  ∂qt∂t_peak  = s.∂qt∂t_peak
+  ∂θ∂t_peak   = s.∂θ∂t_peak
+  k̂           = vertical_unit_vector(atmos.orientation, aux)
+
+  # Piecewise profile for advective moisture forcing
+  P     = air_pressure(TS)
+
+  # Thermodynamic state identification
+  q_pt  = PhasePartition(TS)
+  cvm   = cv_m(TS)
+  
+  linscale_moisture = (z-zl_moisture)/(zh_moisture-zl_moisture)
+  linscale_temp     = (z-zl_sub) / (z_max-zl_sub)
+  linscale_sub      = (z-zl_sub) / (zh_sub-zl_sub)
```

Comment:
I would suggest writing these expressions explicitly in the code below. For example, subsidence has two linear scales but we are only prescribing one, which I think makes the code hard to read. But these are details - up to you.